src/messages.rs

```diff
@@ -43,57 +67,55 @@ pub struct Message {
 /// Defines a transport-related properties that determines how the message
 /// will be broadcast.
 /// TODO: This should be moved out of consensus message definition.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct TransportData {
     pub height: u8,
     pub src_addr: String,
 }
 
 impl Serializable for Message {
-    /// Support serialization for messages that are sent on the wire.
-    fn to_bytes(&self) -> Vec<u8> {
-        let payload_as_vec = match &self.payload {
-            Payload::NewBlock(p) => p.to_bytes(),
-            Payload::Reduction(p) => p.to_bytes(),
-            Payload::Agreement(p) => p.to_bytes(),
-            Payload::AggrAgreement(p) => p.to_bytes(),
-            _ => vec![], // non-serialziable messages are those which are not sent on the wire.
-        };
-
-        let mut buf = BytesMut::with_capacity(payload_as_vec.len());
-        buf.put(&self.header.to_bytes()[..]);
-        buf.put(&payload_as_vec[..]);
-        buf.to_vec()
+    fn write<W: Write>(&self, w: &mut W) -> io::Result<()> {
+        self.header.write(w)?;
+
+        match &self.payload {
+            Payload::NewBlock(p) => p.write(w),
```

Comment:
Are these `write` returning a `Result`? if so you should add the `?`

Reply:
No need for `?` here.  I return `write` result instead.